dev/tools/gen_date_localizations.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart' as args;
+import 'package:path/path.dart' as path;
+
+const String _kCommandName = 'gen_date_localizations.dart';
+
+Future<Null> main(List<String> rawArgs) async {
+  final bool writeToFile = (new args.ArgParser()..addFlag('write', abbr: 'w', defaultsTo: false)).parse(rawArgs)['write'];
+  final bool isRepoRoot = new Directory('.git').existsSync();
+
+  if (!isRepoRoot) {
+    _fatal(
+      '$_kCommandName must be run from the root of the Flutter repository. The '
+      'current working directory is: ${Directory.current.path}'
+    );
+  }
+
+  final File dotPackagesFile = new File(path.join('packages', 'flutter', '.packages'));
+  final bool dotPackagesExists = dotPackagesFile.existsSync();
+
+  if (!dotPackagesExists) {
+    _fatal(
+      'File not found: ${dotPackagesFile.path}. $_kCommandName must be run '
+      'after a successful "flutter update-packages".'
+    );
+  }
+
+  final String pathToIntl = dotPackagesFile
+    .readAsStringSync()
+    .split('\n')
+    .firstWhere(
+      (String line) => line.startsWith('intl:'),
+      orElse: () {
+        _fatal('intl dependency not found in ${dotPackagesFile.path}');
+      },
+    )
+    .split(':')
+    .last;
+
+  final Directory dateSymbolsDirectory = new Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'symbols'));
+  final Map<String, File> symbolFiles = _listIntlData(dateSymbolsDirectory);
+  final Directory datePatternsDirectory = new Directory(path.join(pathToIntl, 'src', 'data', 'dates', 'patterns'));
+  final Map<String, File> patternFiles = _listIntlData(datePatternsDirectory);
+  final List<String> materialLocales = _materialLocales().toList();
+  final StringBuffer buffer = new StringBuffer();
+
+  buffer.writeln(
+'''
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file has been automatically generated.  Please do not edit it manually.
+// To regenerate run (omit -w to print to console instead of the file):
+// dart dev/tools/gen_date_localizations.dart -w
+
+'''
+);
+  buffer.writeln('const Map<String, dynamic> dateSymbols = const <String, dynamic> {');
+  symbolFiles.forEach((String locale, File data) {
+    if (materialLocales.contains(locale))
+      buffer.writeln(_jsonToMapEntry(locale, JSON.decode(data.readAsStringSync())));
```

Comment:
It would be nice to declare which fields we actually need and only write those. That would also give us the opportunity to verify that the fields we expect are still defined.